src/main/java/crypto/AlgorithmAES.java

```diff
@@ -64,4 +71,20 @@ private SecretKeySpec getSecretKeySpec(String secret) {
 		return new SecretKeySpec(key, "AES");
 	}
 
+	@Override
+	public AlgorithmType getType() {
```

Comment:
we could probably replace this with a method public boolean isSymmetric() - it's unlikely that we'll see additional optionen beyond symmetric/asymmetric in the near future ;)